<Type Name="RetryLimitExceededException" FullName="Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException">
  <TypeSignature Language="C#" Value="public sealed class RetryLimitExceededException : Exception" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed beforefieldinit RetryLimitExceededException extends System.Exception" />
  <TypeSignature Language="DocId" Value="T:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class RetryLimitExceededException&#xA;Inherits Exception" />
  <TypeSignature Language="F#" Value="type RetryLimitExceededException = class&#xA;    inherit Exception" />
  <AssemblyInfo>
    <AssemblyName>Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling</AssemblyName>
    <AssemblyVersion>6.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute FrameworkAlternate="azure-dotnet">
      <AttributeName>System.Obsolete("You should use cancellation tokens or other means of stoping the retry loop.")</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
            The special type of exception that provides managed exit from a retry loop. The user code can use this
            exception to notify the retry policy that no further retry attempts are required.
            </summary>
    <remarks>To be added.</remarks>
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public RetryLimitExceededException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling</AssemblyName>
        <AssemblyVersion>6.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
            Initializes a new instance of the <see cref="T:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" /> class with a default error message.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public RetryLimitExceededException (Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException.#ctor(System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (innerException As Exception)" />
      <MemberSignature Language="F#" Value="new Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException : Exception -&gt; Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" Usage="new Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException innerException" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling</AssemblyName>
        <AssemblyVersion>6.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="innerException">The exception that is the cause of the current exception.</param>
        <summary>
            Initializes a new instance of the <see cref="T:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" /> class with a reference to the inner exception
            that is the cause of this exception.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public RetryLimitExceededException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="F#" Value="new Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException : string -&gt; Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" Usage="new Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling</AssemblyName>
        <AssemblyVersion>6.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">The message that describes the error.</param>
        <summary>
            Initializes a new instance of the <see cref="T:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" /> class with a specified error message.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public RetryLimitExceededException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="F#" Value="new Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException : string * Exception -&gt; Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" Usage="new Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling</AssemblyName>
        <AssemblyVersion>6.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">The message that describes the error.</param>
        <param name="innerException">The exception that is the cause of the current exception.</param>
        <summary>
            Initializes a new instance of the <see cref="T:Microsoft.Practices.EnterpriseLibrary.TransientFaultHandling.RetryLimitExceededException" /> class with a specified error message and inner exception.
            </summary>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>